src/sql/src/query_model/hir_generator.rs

```diff
@@ -0,0 +1,125 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+//! Generates a Query Graph Model from a [HirRelationExpr].
+
+use itertools::Itertools;
+
+use crate::plan::HirScalarExpr;
+use crate::query_model::{
+    BoxId, BoxType, Column, ColumnReference, Expr, Model, QuantifierType, Values,
+};
+
+use crate::plan::expr::HirRelationExpr;
+
+pub struct FromHir {
+    model: Model,
+}
+
+impl FromHir {
```

Comment:
Nit: You could consider implementing `From<&HirRelationExpr> for Model`, which would present the code in this file in a more intuitive way to the user.

Reply:
Done. Thank you!